### DISCUSSION TITLE
firecfg: spectacle not taking screenshots even after firecfg --clean

### DISCUSSION THREAD

**Discussion by mystiquewolf** [netblue30/firejail#5063] [Q&A] [answered]
Hello, I've apparently disabled Firejail - I issued `firecfg --clean`, I've disabled and removed the AppArmor `firejail` profile and restarted, there's no active or disabled firejail AppArmor profile now. I've also uninstalled Firejail completely.

With all that said, spectacle, when it opens, shows the error `Could not take a screenshot.`. I launch fireshot in two ways:

1. Via the `Prnt Scrn` button - doesn't work with the above error.
2. From terminal by running `/usr/bin/spectacle` - doesn't work with the above error.

Is there something that I'm missing?

I've opened the following bug report: https://bugs.kde.org/show_bug.cgi?id=451242

**Comment by rusty-snake** [comment, marked as answer]
Is there a spectracle .desktop file in `~/.local/share/applications`? Try to remove it.

Related #5016.

**Reply by mystiquewolf** [reply]
Yes, removing it fixed it! Strange as I was not using the .desktop but was launching directly using the terminal, so I think the command inside the .desktop file should not have executed..

**Reply by rusty-snake** [reply]
firecfg removes `DBusActivable=true` from .desktop files.

**Comment by aardbol** [comment]
Removing the .desktop file is the solution but it keeps reappearing. Is firecfg recreating it?

**Reply by rusty-snake** [reply]
Exactly.

Recent change here (merged 3 days ago): #6153

**Reply by aardbol** [reply]
I installed git version (from arch AUR) and I still have the same issue. It looked like the only solution as to disable spectacle in firecfg to solve this issue, but doing that still creates the desktop file.

**Reply by rusty-snake** [reply]
So you installed firejail from git version, added spectacle to firecfgs ignorelist/removed it from the do-list, deleted the old .desktop file, rerun firecfg, restarted the session and spectacle failed with the same error?

**Reply by aardbol** [reply]
I haven't checked any error, but the issue ended up being the same: I couldn't start spectacle with the prtsc button because of that desktop file. Removing it makes it work again

**Reply by rusty-snake** [reply]
@kmk3 the changes from #6153 work for you, right?

**Comment by kmk3** [comment]
@rusty-snake [on Mar 6](https://github.com/netblue30/firejail/discussions/5063#discussioncomment-8696999):

> the changes from #6153 work for you, right?

Yes.

To ignore a .desktop file you need to use its exact name (without the
`.desktop` suffix).

For example, if the file is called `org.kde.spectacle.desktop`:

/etc/firejail/firecfg.d/my.conf:

```
!org.kde.spectacle
```

To ignore both the main executable and the .desktop file:

```
!org.kde.spectacle
!spectacle
```

**Reply by kmk3** [reply]
Related PR:

* #6268

**Comment by ghost** [comment]
@mystiquewolf Have you trued the suggestions from https://github.com/netblue30/firejail/issues/5127#issuecomment-1715299608 yet?
